portfolio/src/main/webapp/app.js

```diff
@@ -41,60 +41,69 @@ function getBookInfo() {
-        var author = info.authors;
-        var language = info.language;
-        var description = info.description; 
-        var infoLink = info.infoLink;
-        var pageCount = info.pageCount;
-        var publishedDate = info.publishedDate;
-        var publisher = info.publisher;
-        var maturityRating = info.maturityRating;
- 
-        var fieldContentArray = [title, genre, categories, author, language, description, infoLink, pageCount, publishedDate, publisher, maturityRating];
-        var fieldsString;
-        var index = 0;
- 
-        //passing values through as URL parameters
-        for (var field in bookFields){
-           if(index == 0){
-             fieldsString = "?"+field+"="+fieldContentArray[index];
-           }else if(index<bookFields.length-1){
-             fieldsString += "&"+field+"="+fieldContentArray[index]
-           }
-           index ++;
-        }
-        
-        fetch("/data-upload"+fieldsString,{method:"POST"});
- 
-   })
-   .fail(function (jqxhr, status, errorMessage)
-   {
-       $("#bookresult").html("Status Code: " + status+"<br>Error Message: "+errorMessage);
-   }); 
+
+const bookFields =
+    {
+      TITLE: 'title',
+      GENRE: 'genre',
+      CATEGORIES: 'categories',
+      AUTHOR: 'author',
+      LANGUAGE: 'language',
+      DESCRIPTION: 'description',
+      INFO_LINK: 'infoLink',
+      PAGE_COUNT: 'pageCount',
+      PUBLISH_DATE: 'publishedDate',
+      PUBLISHER: 'publisher',
+      MATURITY_RATING: 'maturityRating'
+    }
+
+function searchBooks(book, url) {
+  $.getJSON(url, function(json) {
+     var info = json.items[0].volumeInfo;
+     var title = book.title;
+     var reviews = book.reviews;
+     var genre = book.genre;            // List of genres from GoodReads
+     var categories = info.categories;  // List of genres from Books API
+     var author = info.authors;
+     var language = info.language;
+     var description = info.description;
+     var infoLink = info.infoLink;
+     var pageCount = info.pageCount;
+     var publishedDate = info.publishedDate;
+     var publisher = info.publisher;
+     var maturityRating = info.maturityRating;
+
+     var fieldContentArray = [
+       title, genre, categories, author, language, description, infoLink,
+       pageCount, publishedDate, publisher, maturityRating
+     ];
+     var fieldsString;
+     var index = 0;
+
+     // passing values through as URL parameters
+     for (var field in bookFields) {
+       if (index == 0) {
+         fieldsString = '?' + field + '=' + fieldContentArray[index];
+       } else if (index < bookFields.length - 1) {
+         fieldsString += '&' + field + '=' + fieldContentArray[index]
+       }
+       index++;
+     }
+
+     fetch('/data-upload' + fieldsString, {method: 'POST'});
+   }).fail(function(jqxhr, status, errorMessage) {
+    $('#bookresult')
+        .html('Status Code: ' + status + '<br>Error Message: ' + errorMessage);
+  });
 }
 
+function
```

Comment:
I think we ought to create an AngularJS controller here like we do in our other components and move this new code inside of that controller.